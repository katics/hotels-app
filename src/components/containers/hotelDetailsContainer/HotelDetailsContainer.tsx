import { FC, useEffect } from "react";
import { useHistory, useParams } from "react-router-dom";
import { fetchHotelDetails } from "../../../store/hotel/hotelActions";
import { hotelsSelector } from "../../../store/hotel/hotelSelector";
import {
  useDispatchHotel,
  useSelectorHotel,
} from "../../../store/HotelProvider";
import HotelDetailsLayout from "../../layout/hotelDetailsLayout/HotelDetailsLayout";
import Spinner from "../../spinner/Spinner";

interface Params {
  hotelId: string;
}

const HotelDetailsPage: FC = () => {
  //const history = useHistory();
  const { hotelDetails, isLoading } = useSelectorHotel(hotelsSelector);
  const hotelDispatch = useDispatchHotel();

  const { hotelId } = useParams<Params>();

  useEffect(() => {
    hotelDispatch(fetchHotelDetails(hotelId));
  }, [hotelId]);
  return (
    <>
      {isLoading ? (
        <Spinner />
      ) : (
        <HotelDetailsLayout
          name={hotelDetails.name}
          city={hotelDetails.city}
          country={hotelDetails.country}
          stars={hotelDetails.stars}
          image={hotelDetails.image}
          likes={hotelDetails.likes}
          description={hotelDetails.description}
          price={hotelDetails.price}
        />
      )}
    </>
  );
};
export default HotelDetailsPage;
