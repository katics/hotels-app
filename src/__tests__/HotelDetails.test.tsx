import HotelDetailsContainer from "../components/containers/hotelDetailsContainer/HotelDetailsContainer";
import { StaticRouter } from "react-router-dom";

import { render } from "@testing-library/react";

describe("Hotel Details Page", () => {
  it("Should match hotel details page snapshot", () => {
    const { container } = render(
      <StaticRouter>
        <HotelDetailsContainer />
      </StaticRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
