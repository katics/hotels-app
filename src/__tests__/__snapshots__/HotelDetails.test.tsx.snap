// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotel Details Page Should match hotel details page snapshot 1`] = `
<div
  class="spinner-border text-primary spinner-border-lg"
/>
`;
